x-pack/plugins/enterprise_search/public/applications/app_search/components/search_ui/search_ui.tsx

```diff
@@ -7,87 +7,79 @@
+import { AppSearchPageTemplate } from '../layout';
 
+import { EmptyState } from './components/empty_state';
 import { SearchUIForm } from './components/search_ui_form';
 import { SearchUIGraphic } from './components/search_ui_graphic';
 import { SEARCH_UI_TITLE } from './i18n';
 import { SearchUILogic } from './search_ui_logic';
 
 export const SearchUI: React.FC = () => {
   const { loadFieldData } = useActions(SearchUILogic);
+  const { engine } = useValues(EngineLogic);
 
   useEffect(() => {
     loadFieldData();
   }, []);
 
   return (
-    <>
-      <SetPageChrome trail={getEngineBreadcrumbs([SEARCH_UI_TITLE])} />
-      <EuiPageHeader pageTitle={SEARCH_UI_TITLE} />
-      <FlashMessages />
-      <EuiPageContentBody>
-        <EuiFlexGroup alignItems="flexStart">
-          <EuiFlexItem>
-            <EuiText size="s" color="subdued">
-              <p>
-                <FormattedMessage
-                  id="xpack.enterpriseSearch.appSearch.engine.searchUI.bodyDescription"
-                  defaultMessage="Search UI is a free and open library for building search experiences with React. {link}."
-                  values={{
-                    link: (
-                      <EuiLink target="_blank" href="https://github.com/elastic/search-ui">
-                        <FormattedMessage
-                          id="xpack.enterpriseSearch.appSearch.engine.searchUI.repositoryLinkText"
-                          defaultMessage="Learn more"
-                        />
-                      </EuiLink>
-                    ),
-                  }}
-                />
-              </p>
-              <p>
-                <FormattedMessage
-                  id="xpack.enterpriseSearch.appSearch.engine.searchUI.lowerBodyDescription"
-                  defaultMessage="Use the fields below to generate a sample search experience built with Search UI. Use the sample to preview search results, or build upon it to create your own custom search experience. {link}."
-                  values={{
-                    link: (
-                      <EuiLink target="_blank" href={`${DOCS_PREFIX}/reference-ui-guide.html`}>
-                        <FormattedMessage
-                          id="xpack.enterpriseSearch.appSearch.engine.searchUI.guideLinkText"
-                          defaultMessage="Learn more"
-                        />
-                      </EuiLink>
-                    ),
-                  }}
-                />
-              </p>
-            </EuiText>
-            <EuiSpacer />
-            <SearchUIForm />
-          </EuiFlexItem>
-          <EuiFlexItem>
-            <SearchUIGraphic />
-          </EuiFlexItem>
-        </EuiFlexGroup>
-      </EuiPageContentBody>
-    </>
+    <AppSearchPageTemplate
+      pageChrome={getEngineBreadcrumbs([SEARCH_UI_TITLE])}
+      pageHeader={{ pageTitle: SEARCH_UI_TITLE }}
+      isEmptyState={Object.keys(engine.schema || {}).length === 0}
```

Comment:
Heads up, I fully intend on refactoring this `Object.keys` check to an EngineLogic selector - but that's going to come in a follow-up polish PR once all the views land. Essentially I want to tackle the concept of empty engine polling more holistically and at the engine logic level, and ensure _all_ pages update dynamically when an engine is empty and receives new documents/schema.

Reply:
Also, I did check w/ Davey on this and he confirmed that this view should check for empty schema and not just for empty documents. So someone could index a document and create a schema but then delete the document, and this would no longer show the empty state (which matches how Relevance Tuning & Result Settings behave)